libs/framework/include/celix_bundle_context.h

```diff
@@ -126,8 +126,34 @@ typedef struct celix_service_registration_options {
      * for this.
      */
     const char *serviceVersion OPTS_INIT;
+
+    /**
+     * If set to > 0, this svc id will be used to register the service.
+     * The reservedSvcId should be reserved with a call to celix_bundleContext_reserveSvcId
+     */
+    long reservedSvcId OPTS_INIT;
 } celix_service_registration_options_t;
 
+/**
+ * Reserves a service id, which is expected to be used to register a service in the future.
+ *
+ * If a celix_bundleContext_unregisterService with the reserved service id is called earlier than
+ * the celix_bundleContext_registerServiceWithOptions with the reserved service. The registering
+ * of the service will be cancelled instead.
```

Comment:
What happens on subsequent calls to register? It is always and forever cancelled? If so, how to reclaim a svc Id?

Reply:
it always cancelled. subsequent call will return an error code (and log).
You cannot reclaim the svc id.

Reply:
I think this needs to be added to the documentation.